tools/utils/policy_test

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+import os
+import argparse
+import json
+import yaml
+import tests.common
+import c7n.credentials
+import c7n.utils
+import placebo
+import shutil
+import boto3
+
+parser = argparse.ArgumentParser('Policy Tester')
+parser.add_argument('--assume', dest='assume_role', help='Role to assume')
+subparsers = parser.add_subparsers(dest='command', help='Execution mode')
+subparsers.required=True
+
+record_parser = subparsers.add_parser('record')
+record_parser.add_argument('--cache', type=str, default='cache', help='Cache dir to write to')
+record_parser.add_argument('policy', type=argparse.FileType('r'), help='Policy file to consume')
+record_parser.add_argument('name', type=str, help='Name of the policy')
+
+test_parser = subparsers.add_parser('replay')
+test_parser.add_argument('--cache', type=str, default='cache' help='Cache dir to consume')
+test_parser.add_argument('policy', type=argparse.FileType('r'), help='Policy file to consume')
+test_parser.add_argument('name', type=str, help='Name of the policy')
+
+args = parser.parse_args()
+
+class TestPolicy(tests.zpill.PillTest):
+    def __init__(self, policy, cache, assume_role=None):
+        self.policy = policy
+        self.assume_role = assume_role
+        self.account_id = None
+        self.cleanup = []
+        self.placebo_dir = cache
+        self.test_dir = os.path.join(self.placebo_dir, self.policy['name'])
+        if assume_role:
+            self.account_id = self.assume_role.split(':')[4]
```

Comment:
Same issue as [here](https://github.com/cloud-custodian/cloud-custodian/issues/1658)?

Reply:
not per se, it wasn't really clear what that user was doing since they were attempting to initialize things independently of custodian, but yeah it was related to not having account id steup.